scripts/visualize/visualize_utils.R

```diff
@@ -151,25 +156,31 @@ clip_sp.SpatialLines <- function(sp, xlim, ylim, ...){
 clip_sp.Spatial <- function(sp, xlim, ylim, ..., clip.fun = rgeos::gIntersection){
   
   clip <- as.sp_box(xlim, ylim, sp::CRS(sp::proj4string(sp)))
-  g.i <- rgeos::gIntersects(sp, clip, byid = T) 
-  
-  sp <- rgeos::gBuffer(sp, byid=TRUE, width=0) # zero buffer to avoid TopologyException
-  has.data <- ("data" %in% slotNames(sp))
-
-  out <- lapply(which(g.i), function(i) {
-    g.out <- clip.fun(sp[i,], clip)
-    row.names(g.out) <- row.names(sp)[i]
-    return(g.out)
-  })
   
-  # use rbind.SpatialPolygons method to combine into a new object.
-  clipped.sp <- do.call("rbind", out)
   
-  if(has.data){
-    data.out <- as.data.frame(sp)[g.i, ]
-    row.names(data.out) <- row.names(sp)[g.i]
-    clipped.sp <- as(object = clipped.sp, Class = class(sp))
-    clipped.sp@data <- data.out
+  if (!inherits(sp, 'SpatialPoints')){
+    g.i <- rgeos::gIntersects(sp, clip, byid = T) 
+    has.data <- ("data" %in% slotNames(sp))
+    
+    out <- lapply(which(g.i), function(i) {
+      g.out <- clip.fun(sp[i,], clip)
+      row.names(g.out) <- row.names(sp)[i]  
+      return(g.out)
+    })
+    
+    # use rbind.SpatialPolygons method to combine into a new object.
+    clipped.sp <- do.call("rbind", out)
+    
+    if(has.data){
```

Comment:
I think this could also just check if it's of class "SpatialPointsDataFrame"?

Reply:
Yes, I need to refactor this. I naively thought I could write one method that worked for all of the spatial classes.. but no...